ui/src/app/app.tsx

```diff
@@ -124,24 +122,69 @@ export class App extends React.Component<{}, {version?: Version; popupProps: Pop
             <Provider value={providerContext}>
                 {this.state.popupProps && <Popup {...this.state.popupProps} />}
                 <Router history={history}>
-                    <Switch>
-                        <Redirect exact={true} path={uiUrl('')} to={workflowsUrl} />
-                        <Redirect from={timelineUrl} to={uiUrl('workflows')} />
+                    <Layout navItems={navItems} version={() => <>{this.state.version ? this.state.version.version : 'unknown'}</>}>
+                        <Notifications notifications={this.notificationsManager.notifications} />
+
                         <ErrorBoundary>
-                            <Layout navItems={navItems} version={() => <>{this.state.version ? this.state.version.version : 'unknown'}</>}>
-                                <Notifications notifications={this.notificationsManager.notifications} />
-                                {Object.keys(routes).map(path => {
-                                    const route = routes[path];
-                                    return <Route key={path} path={path} component={route.component} />;
-                                })}
-                            </Layout>
+                            <Switch>
+                                <Route exact={true} strict={true} path={uiUrl('')}>
+                                    <Redirect to={workflowsUrl} />
+                                </Route>
+                                <Route exact={true} strict={true} path={timelineUrl}>
+                                    <Redirect to={workflowsUrl} />
+                                </Route>
+                                {this.state.namespace && (
```

Comment:
these are the redirects, it seems you should use `exact` and `strict` if you can